File: ACM/Acwing/3288.cpp
/*
 * @Author: NEFU AB-IN
 * @Date: 2022-03-18 16:29:32
 * @FilePath: \ACM\Acwing\3288.cpp
 * @LastEditTime: 2022-03-18 16:29:33
 */
#include <algorithm>
#include <cstring>
#include <iostream>
#include <unordered_map>

using namespace std;

typedef long long LL;

unordered_map<int, LL> A, B;

int n, a, b;
int id, value;

int main()
{
    scanf("%d%d%d", &n, &a, &b);
    for (int i = 0; i < a; i++)
    {
        scanf("%d%d", &id, &value);
        A[id] = value;
    }
    for (int i = 0; i < b; i++)
    {
        scanf("%d%d", &id, &value);
        B[id] = value;
    }
    LL ans = 0;
    for (auto [id, value] : A)
    {
        if (B[id])
        {
            ans += A[id] * B[id];
            A[id] = 0;
            B[id] = 0;
        }
    }
    printf("%lld\n", ans);
    return 0;
}

File: ACM/Acwing/3288/3288.py
'''
Author: NEFU AB-IN
Date: 2022-03-18 15:45:29
FilePath: \ACM\Acwing\3288.py
LastEditTime: 2022-03-18 16:23:38
'''
from collections import Counter

A = Counter()
B = Counter()

n, a, b = map(int, input().split())

for i in range(a):
    index, value = map(int, input().split())
    A[index] = value
for i in range(b):
    index, value = map(int, input().split())
    B[index] = value

ans = 0
for index in A.keys():
    if B[index]:
        ans += A[index] * B[index]

print(ans)